guides/LIBRARIES.md

```diff
@@ -0,0 +1,125 @@
+Library choices
+==================
+
+This is a partially-complete list of libraries used for different areas by Digital Publishing.
+
+The aims of this document are to define standards around what libraries are being used, provide a place to center conversation about changing libraries and be a reference point to the existance of internal libraries.
+
+## Table of Contents
+1. [Internal Library Standards](#Internal-Library-Standards)
+2. [HTTP and Middleware](#HTTP-and-middleware)
+3. [Config & Logging](#config-&-logging)
+4. [Database Drivers](#database-drivers)
+5. [Testing](#testing)
+6. [Misc.](#misc.)
+
+# Internal Library Standards
+
+We create internal libraries where we want our interaction with a 3rd-party library to be
+standardized across all of our services. Often this is as simple as adding our health 
+checking ability and surfacing the behaviour that's relevant to us, in other cases it 
+provides us the opportunity to set defaults consistently across our micro services.
+
+Our internal libraries typically begin with `dp-` in the repo name, and *must* reside in the
+`ONSdigital` Github organization. 
+  * Go note: It is acceptable for files to live at the top level of these repos and take the single word name after `dp-` as the package name. e.g. `dp-kafka` uses the `package kafka` definiton
+
+These client libraries all serve that high level wrapping purpose:
+  * [dp-api-clients-go](https://github.com/ONSdigital/dp-api-clients-go) - which houses all of our API clients
+  * [dp-elasticsearch](https://github.com/ONSdigital/dp-elasticsearch)
+  * [dp-nomad](https://github.com/ONSdigital/dp-nomad)
+  * [dp-vault](https://github.com/ONSdigital/dp-vault)
+  * [dp-mongodb](https://github.com/ONSdigital/dp-mongodb)
+  * [dp-s3](https://github.com/ONSdigital/dp-s3)
+  * [dp-kafka](https://github.com/ONSdigital/dp-kafka)
+
```

Comment:
For java there's also `dp-dataset-api-java-client`

Reply:
Also:

`dp-graph`
`dp-reporter-client` <- maybe this should be moved to dp-api-clients-go
`dp-frontend-models` - see Dans comment above
`dp-ssqs`

Reply:
`go-ns/avro` is important for serialisation of kafka messages on the cmd backend pipeline (arguably should be moved to kafka library or it's own.

Reply:
Thanks, I've added the ones I'd missed 👍 

with the reporter client, its not actually an HTTP client to my understanding, its queueing kafka messages, so I don't think that should be moved to API clients. I'm not sure if client is the best term for this or not, but im not too worried about it as long as its a standalone thing